tools/cloud-build/daily-tests/blueprints/ml-gke-e2e.yaml

```diff
@@ -57,7 +57,9 @@ deployment_groups:
       name: g2-latest-driver
       machine_type: g2-standard-4
       guest_accelerator:
-      - gpu_driver_installation_config:
+      - type: nvidia-tesla-t4
+        count: 2
```

Comment:
Is this allowed, I thought we could add the type of GPU only for `n series` machine families ?   
Did you mean `nvidia-l4` ? 
Also just for context, how is this change important to this PR ?

Reply:
Context: https://github.com/GoogleCloudPlatform/cluster-toolkit/pull/3024#issuecomment-2346525779

Wanted to see if the any `type`, `count` keys help with the breaking test. 

You are right about the compatibility of the machine type and accelerator. Will use `nvidia-l4`

Reply:
Accelerator type is meant to be optional for a known machine family, i.e. our code should be able to extract that `g2-standard-4` corresponds to `1 nvidia-l4` . 

It now seems that based on this change, we are regressing that feature.

Reply:
> It now seems that based on this change, we are regressing that feature.

Right. 

> .. i.e. our code should be able to extract that g2-standard-4 corresponds to 1 nvidia-l4 .

That was my understanding as well. I suspect there might be a bug here https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/develop/modules/compute/gke-node-pool/gpu_definition.tf#L56

When the `var.guest_accelerator` is specified it overrides `local.generated_guest_accelerator`. But, if the specified value does not have type, count then the `local.guest_accelerator` won't have those keys either.

So, we either fix the gpu_definition itself to have those keys or the `reservation_vm_properties` map construction handles it. I will form an opinion and made changes accordingly.